src/Generators/Shared/RoslynExtensions.cs

```diff
@@ -0,0 +1,203 @@
+using System;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+#pragma warning disable CA1716
+namespace Microsoft.Gen.Shared;
+#pragma warning restore CA1716
+
+internal static class RoslynExtensions
+{
+    // Copied from: https://github.com/dotnet/roslyn/blob/main/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/CompilationExtensions.cs
```

Comment:
I recommend using permalinks because there's no guarantee the file or its content will always exist in the main
```suggestion
    // Copied from: https://github.com/dotnet/roslyn/blob/dd13c33ad54aa8afed4442b029bb14cb5893bc7e/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/CompilationExtensions.cs
```

Reply:
changed all to use permalinks